### ISSUE TITLE
Electrum throws exception if network is not available when creating wallet

### ISSUE THREAD

**Issue by thunderbiscuit** [bitcoindevkit/bdk-ffi#135]
When attempting to create a wallet without a network connection, bdk-ffi throws an exception:
```sh
Caused by: org.bitcoindevkit.BdkException$Electrum: Electrum(IOError(Custom { kind: Uncategorized, error: "failed to lookup address information: No address associated with hostname" }))
```

You can reproduce the error by building the [simple-wallet](https://github.com/thunderbiscuit/devkit-wallet) branch of the Devkit Wallet.

Overall this feels like a bug, because while we'd expect methods like `sync` and `broadcast` to throw when there is no network connection, simply creating a wallet should not require a network connection, even if we're building only `OnlineWallet`s with bdk-ffi at the moment.

**Comment by notmandatory** [comment]
I figured out the issue, when you setup the blockchain it does a DNS lookup on the URL to get the correct IP address.. this requires a network connection.  One work around would be to use a hard coded IP address for the blockstream server but this isn't great since they could change the IP and then things would break. 

The `onlineWalletInMemory()` test in `bdk-kotlin` with the network turned off is a quick way to repro the issue. Or by putting a bogus domain name even with the network on gives the same error.
```kotlin
    @Test
    fun onlineWalletInMemory() {
        val database = DatabaseConfig.Memory
        val blockchain = BlockchainConfig.Electrum(
            ElectrumConfig(
                "ssl://electrum.blockstream.info:60002",
                null,
                5u,
                null,
                100u
            )
        )
        val wallet = Wallet(descriptor, null, Network.TESTNET, database, blockchain)
        assertNotNull(wallet)
        val network = wallet.getNetwork()
        assertEquals(network, Network.TESTNET)
    }
```

**Comment by notmandatory** [comment]
This should be fixed when #143 is completed.

**Comment by notmandatory** [comment]
Fixed by #149
